### PR TITLE
Add helm chart

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

**Comment by corest** [comment]
I expected to see some usual helm staff we have in other repos. But instead, I see a bunch of customization. Is this a new way to manage helm values?

File: config/components/psp/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: kustomize.config.k8s.io/v1alpha1
```
Comment (kopiczko) [inline config/components/psp/kustomization.yaml:1]: This psp component is our standard way to define PSPs. This 100% copy paste from https://github.com/giantswarm/capi-migration/tree/b452e8bed1516d0cca989c8e8183198067db791e/config/components/psp. This is cool kustomize feature.

File: Makefile [hunk 2]
```diff
@@ -71,11 +76,15 @@ install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~
 uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/crd | kubectl delete -f -
 
+deploy: NAME_SUFFIX ?= $(USER)
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	cd config/dev && $(KUSTOMIZE) edit set image controller=${IMG}
+	cd config/dev && $(KUSTOMIZE) edit set namesuffix -- -$(NAME_SUFFIX)
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
```
Comment (kopiczko) [inline Makefile:83]: So everybody can `make deploy` and that creates resources suffixed with your current `$USER`.

**Comment by kopiczko** [comment]
> I expected to see some usual helm staff we have in other repos. But instead, I see a bunch of customization. Is this a new way to manage helm values?

At the end of the day everything lands in `helm/management-cluster-admission/templates/kustomize-out.yaml` (generated) when `make manifests` is being run.

Kubebuilder uses kustomize, so I think it's beneficial to start getting into this especially since we can generate valid helm charts which are on par with what we have right now.

**Review by corest — APPROVED** [review]
> Kubebuilder uses kustomize, so I think it's beneficial to start getting into this especially since we can generate valid helm charts which are on par with what we have right now.

I'm fine with that. Just wanted to know if there was some discussion or decision has been done to use more kustomize now. And also what it covers - operators only or all the projects?

**Comment by kopiczko** [comment]
> I'm fine with that. Just wanted to know if there was some discussion or decision has been done to use more kustomize now. And also what it covers - operators only or all the projects?

Let's continue in Slack: https://gigantic.slack.com/archives/C010L893ZFW/p1617879384015500?thread_ts=1617877651.015100&cid=C010L893ZFW

**Pull request merged**